packages/cli/src/databases/repositories/mixins/queryMethods.mixin.ts

```diff
@@ -0,0 +1,48 @@
+import { jsonParse } from 'n8n-workflow';
+import type { Constructor } from '@/Interfaces';
+import * as utils from '@/utils';
+import type { Schema } from '../schema';
+
+function hasFieldNamesGetter(schema: unknown): schema is { getFieldNames(): string[] } {
+	return (schema as Schema)?.getFieldNames !== undefined;
+}
+
+export function mixinQueryMethods<T extends Constructor>(base: T) {
+	class Derived extends base {
+		static toQueryFilter(rawFilter: string, schema: typeof Schema) {
+			const asObj = jsonParse(rawFilter, { errorMessage: 'Failed to parse filter JSON' });
+
+			const validFilter = new schema(asObj);
+
+			return Object.fromEntries(
+				Object.keys(validFilter).map((field: keyof Schema) => [field, validFilter[field]]),
+			);
+		}
+
+		static toQuerySelect(rawSelect: string, schema: typeof Schema) {
+			const asArr = jsonParse(rawSelect, { errorMessage: 'Failed to parse select JSON' });
+
+			if (!utils.isStringArray(asArr)) {
+				throw new Error('Parsed select is not a string array');
+			}
+
+			const validSelect = hasFieldNamesGetter(schema)
+				? asArr.filter((f) => schema.getFieldNames().includes(f)) // strip out unknown fields
+				: asArr;
+
+			return validSelect.reduce<Record<string, true>>((acc, field) => {
+				return (acc[field] = true), acc;
+			}, {});
+		}
+
```

Comment:
I'm not sure if these belong in the repositories. Generally parsing of body or query string into a typed object goes in a middleware or a controller.

Reply:
Moved to middleware.